### ISSUE TITLE
Massive memory spike requiring restart

### ISSUE THREAD

**Issue by lamancha212** [tboby/cwtools-vscode#8]
Several times since the latest update something is causing cwtools to use a massive amount of memory (10 gb+) and a great deal of processor usage as well.

This last occurred when I was trying to search inside a file.  I have two windows open with the extension running and files (including the same file) in each if that might help pinpoint the issue.

I had never encountered this before the last update.

Also possibly unrelated (as I also updated vscode), but I used to be able to edit the same file in multiple windows and it would share the edits across them simultaneously (I could make sure they were in sync due to the edit dot).  Now it requires I save to update the changes across windows.  This probably isn't related to your extension but in case it was and you might know of a fix I thought I'd bring it up.

**Comment by tboby** [comment]
Hi!
I would expect high peak memory usage when you first open VS Code and cwtools "activates". This should peak around 2GB, but should drop to below 1GB once loaded. It'll drop even further if memory pressure is applied.

10GB sounds very high, unless you have something on the order of 60-70k script files (which is 4-5x bigger than any total conversion I've seen).

Is this something you can reproduce? If so, it would be very helpful if you could do the following:
1. Enable logging: "File", "Preferences", "Settings", then search for cwtools and set `cwtools.trace.server` to "Messages".
2. Close VS Code
3. Reproduce the problem (opening multiple windows if you need to).
4. For each window, get the logs, "View", "Output" then in the panel that opened select "Paradox language server" in the dropdown on the right.
5. Copy those logs into files and drop them here. You might want to redact your username.

Re your syncing problem: check the setting `files.autoSave`. and `files.autoSaveDelay`
I've never tried opening two copies of cwtools on the same files, but it should be fine (although they'll do twice as much work for the shared files as they don't share). Have you considered using workspaces? They might help in your situation.

**Comment by lamancha212** [comment]
Well I turned on logging and then couldn't reproduce it, but somehow in the process I also managed to turn off error reporting from the extension which explains why the memory usage never even got above 500 mb.

I thought maybe the extension was maybe just recognizing customizable localizations and eventtarget scopes in the localization files without flagging them but I can't get any error to show up now.  I'll try reinstalling and see if it happens once I get that to work again.  Thanks for your reply, it is probably something on my end.
